src/cpp/session/modules/SessionHTMLPreview.cpp

```diff
@@ -964,6 +975,46 @@ void handlePreviewRequest(const http::Request& request,
    }
 }
 
+SEXP rs_showPageViewer(SEXP fileSEXP)
+{
+   try
+   {
+      // get full path to file
+      FilePath filePath = module_context::resolveAliasedPath(r::sexp::safeAsString(fileSEXP));
+
+      // create temp file to write standalone html to (place it within a temp dir
+      // so the default download file name is pretty)
+      FilePath viewerTempDir = module_context::tempFile("page-viewer", "dir");
+      Error error = viewerTempDir.ensureDirectory();
+      if (error)
+         throw r::exec::RErrorException(r::endUserErrorMessage(error));
+      FilePath viewerFilePath = viewerTempDir.childPath(filePath.filename());
+
+      // create base64 encoded version
+      error = module_context::createSelfContainedHtml(filePath, viewerFilePath);
```

Comment:
Do you think this should be capped in some way on file size? (i.e. if there's a limit on the amount of data we can ferry in a client event, or we risk an apparent hang over some size)